library/uri/generic/host_spec.rb

```diff
@@ -2,7 +2,14 @@
 require 'uri'
 
 describe "URI::Generic#host" do
-  it "needs to be reviewed for spec completeness"
+  ruby_version_is "3.2" do
+    it "returns empty string if is not valid" do
+      url = URI.parse('http:////foo.com')
+
+      url.to_s.should == 'http:////foo.com'
```

Comment:
I suppose `#to_s` should be tested in its own test file. It makes sense to have both checks in the more generic `parser` test file. But here we have more granular test cases.

Reply:
Please also note that there are `////` for a `host`-related test, and `///` (3 '/' characters) for a `to_s` related test. Not sure whether it's important but nevertheless...

Reply:
I meant this one - `https://github.com/ruby/spec/blob/master/library/uri/generic/to_s_spec.rb`.